src/io/flutter/project/FlutterIconProvider.java

```diff
@@ -20,7 +21,7 @@
 public class FlutterIconProvider extends IconProvider {
   @Nullable
   public Icon getIcon(@NotNull final PsiElement element, @Iconable.IconFlags final int flags) {
-    if (element instanceof PsiDirectory) {
+    if (element instanceof PsiDirectory && FlutterModuleUtils.hasFlutterModule(element.getProject())) {
```

Comment:
I see your concern.  For this I think checking for a declared Flutter module should suffice.  Updating.